moabb/contexts/motor_imagery.py

```diff
@@ -1,47 +1,35 @@
 """Motor Imagery contexts"""
 
 import numpy as np
-from .base import WithinSubjectContext
+from .base import CrossSubjectContext, WithinSubjectContext, BaseContext
 from mne import Epochs, find_events
 from mne.epochs import concatenate_epochs, equalize_epoch_counts
 from sklearn.model_selection import cross_val_score, LeaveOneGroupOut, KFold
 
+class MotorImageryContext(BaseContext):
+    """Base motor imagery context
 
-class MotorImageryMultiClasses(WithinSubjectContext):
-    """Motor Imagery for multi class classification
-
-    Multiclass motor imagery context. Evaluation is done in Randomized KFold or
-    LeaveOneGroupOut (depending on the group variable, can be run or session)
-    with accuracy as a metric. Epochs count are equalized.
-
-    Parameters
-    ----------
-    datasets : List of Dataset instances.
-        List of dataset instances on which the pipelines will be evaluated.
-    pipelines : Dict of pipelines instances.
-        Dictionary of pipelines. Keys identifies pipeline names, and values
-        are scikit-learn pipelines instances.
-    fmin : float | None, (default 7.)
-        Low cut-off frequency in Hz. If None the data are only low-passed.
-    fmax : float | None, (default 35)
-        High cut-off frequency in Hz. If None the data are only high-passed.
+    Defines motor imagery-specific epoching and preprocessing so that, combined
+    with a child of BaseContext, it is possible to create contexts of interest e.g.
+    MotorImageryMultiClassWithinSubject
 
-    See Also
-    --------
-    MotorImageryTwoClasses
     """
-
     def __init__(self, datasets, pipelines, fmin=7., fmax=35.):
         self.fmin = fmin
         self.fmax = fmax
         super().__init__(datasets, pipelines)
 
     def _epochs(self, dataset, subjects, event_id):
-        """epoch data"""
+        """epoch data
+
+        NOTE: Is this necessary to have defined within the 'motor imagery'
```

Comment:
Indeed. Initially, my intention was to fix parameter in the code (i.e. not having them in arguments) to avoid performance hacking by changing parameters. But now this can probably be reused across many paradigm.

this preprocessing is kind of standard, but i'm wondering if it's not too limiting. But we still have the option to redifine it in the context.